lib/mixlib/log.rb

```diff
@@ -157,5 +157,26 @@ def logger_for(*opts)
       end
     end
 
+    def all_loggers
+      [@logger, *@loggers].uniq
+    end
+
+    # select all loggers with File log devices
+    def loggers_to_close
+      loggers_to_close = []
+      all_loggers.each do |logger|
+        next unless logdev = logger.instance_variable_get(:"@logdev")
```

Comment:
This seems like a bad idea. Needs a hell of a justification as a comment at least.